### ISSUE TITLE
too-few-format-args false positive

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#310]
Originally reported by: **Claudiu Popa (BitBucket: [PCManticore](http://bitbucket.org/PCManticore), GitHub: @PCManticore)**

---

Given the following, pylint emits a too-few-format-args warning. The problem is that the manual position arguments computation doesn't take into account duplicates of arguments (in our example, the 0).

```
#!python

q = """SELECT 'INSERT INTO "{0}" VALUES({1})' FROM "{0}";""".format(
            1,
            ",".join("""'||quote("{0}")||'""".format(col.replace('"', '""')) for col in ["a"]))
```

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/310
